From: wuxiaoyi <xiaoyiwu@ubuntukylin.com>
Date: Mon, 19 Dec 2016 17:53:58 +0800
Subject: exit the loop after trying 20 times

---
 src/module/remote/remote.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/module/remote/remote.c b/src/module/remote/remote.c
index 18ffee3..eda4497 100644
--- a/src/module/remote/remote.c
+++ b/src/module/remote/remote.c
@@ -121,7 +121,7 @@ int CreateSocket(const char *name)
 
 int UdAccept(int listenfd)
 {
-    for (;;) {
+    for (int i =0;i<20;i++) {
         int newsock = 0;
         struct sockaddr_un cliaddr;
         socklen_t len = sizeof(struct sockaddr_un);
@@ -135,6 +135,7 @@ int UdAccept(int listenfd)
 
         return newsock;
     }
+    return -1;
 }
 
 static void SendIMState(FcitxRemote* remote, int fd)
