Index: epiphany-browser-3.18.5/configure.ac
===================================================================
--- epiphany-browser-3.18.5.orig/configure.ac
+++ epiphany-browser-3.18.5/configure.ac
@@ -20,7 +20,7 @@ m4_define([epiphany_version_minor],[18])
 m4_define([epiphany_version_micro],[5])
 m4_define([epiphany_version],[epiphany_version_major.epiphany_version_minor.epiphany_version_micro])
 
-AC_INIT([GNOME Web Browser],[epiphany_version],[http://bugzilla.gnome.org/enter_bug.cgi?product=epiphany],[epiphany])
+AC_INIT([GNOME Web Browser],[epiphany_version],[http://bugzilla.gnome.org/enter_bug.cgi?product=epiphany],[epiphany-browser])
 AX_IS_RELEASE([git-directory])
 
 AC_PREREQ([2.59])
