From: Josselin Mouette <joss@debian.org>
Date: Fri, 2 Jul 2010 23:20:04 +0000
Subject: Install as epiphany-browser not epiphany

---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index a664c5b..31e3307 100644
--- a/meson.build
+++ b/meson.build
@@ -13,9 +13,9 @@ datadir = join_paths(prefix, get_option('datadir'))
 libdir = join_paths(prefix, get_option('libdir'))
 libexecdir = join_paths(prefix, get_option('libexecdir'))
 localedir = join_paths(prefix, get_option('localedir'))
-pkgdatadir = join_paths(datadir, 'epiphany')
-pkglibdir = join_paths(libdir, 'epiphany')
-pkglibexecdir = join_paths(libexecdir, 'epiphany')
+pkgdatadir = join_paths(datadir, 'epiphany-browser')
+pkglibdir = join_paths(libdir, 'epiphany-browser')
+pkglibexecdir = join_paths(libexecdir, 'epiphany-browser')
 webextensionsdir = join_paths(pkglibdir, 'web-extensions')
 
 iso_codes = dependency('iso-codes')
