Index: im-config-0.29/default/im-config
===================================================================
--- im-config-0.29.orig/default/im-config
+++ im-config-0.29/default/im-config
@@ -6,7 +6,9 @@
 #IM_CONFIG_DEFAULT_MODE=cjkv
 
 # Set locale dependent preferred IM over standard auto mode
-IM_CONFIG_PREFERRED_RULE="zh_CN,fcitx:zh_TW,fcitx:zh_HK,fcitx:zh_SG,fcitx"
+if [ "${XDG_CURRENT_DESKTOP##*:}" != 'GNOME' -a "${XDG_CURRENT_DESKTOP%%:*}" != 'GNOME-Flashback' ]; then
+    IM_CONFIG_PREFERRED_RULE="zh_CN,fcitx:zh_TW,fcitx:zh_HK,fcitx:zh_SG,fcitx:ja_JP,fcitx:ko_KR,fcitx:vi_VN,fcitx"
+fi
 
 # User and system wide configuration is normally done via im-config program.
 # The above IM_CONFIG_PREFERRED_RULE sets locale dependent preferred IM
