Index: gnome-bluetooth-3.0.0/configure.ac
===================================================================
--- gnome-bluetooth-3.0.0.orig/configure.ac	2011-05-24 11:45:22.573543108 -0400
+++ gnome-bluetooth-3.0.0/configure.ac	2011-05-24 11:45:33.643543107 -0400
@@ -41,9 +41,6 @@
 # i18n stuff
 IT_PROG_INTLTOOL([0.40.0])
 
-AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.17])
-
 GETTEXT_PACKAGE=gnome-bluetooth2
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE",
