--- a/data/default-bookmarks.rdf.in
+++ b/data/default-bookmarks.rdf.in
@@ -34,4 +34,28 @@
          https://duckduckgo.com/?q=%s&amp;t=epiphany&amp;kl=fi-fi&amp;kad=fi_FI -->
     <_ephy:smartlink>https://duckduckgo.com/?q=%s&amp;t=epiphany</_ephy:smartlink>
   </item>
+  <item rdf:about="http://www.ubuntu.com/">
+    <_title>Ubuntu</_title>
+    <_link>http://www.ubuntu.com/</_link>
+  </item>
+  <item rdf:about="https://wiki.ubuntu.com/">
+    <_title>Ubuntu Wiki (community-edited website)</_title>
+    <_link>https://wiki.ubuntu.com/</_link>
+  </item>
+  <item rdf:about="https://answers.launchpad.net/ubuntu/+addquestion">
+    <_title>Make a Support Request to the Ubuntu Community</_title>
+    <_link>https://answers.launchpad.net/ubuntu/+addquestion</_link>
+  </item>
+  <item rdf:about="http://www.debian.org/">
+    <_title>Debian (Ubuntu is based on Debian)</_title>
+    <_link>http://www.debian.org/</_link>
+  </item>
+  <item rdf:about="https://one.ubuntu.com/">
+    <_title>Ubuntu One - The personal cloud that brings your digital life together</_title>
+    <_link>https://one.ubuntu.com/</_link>
+  </item>
+  <item rdf:about="http://www.gnome.org/">
+    <_title>GNOME</_title>
+    <_link>http://www.gnome.org/</_link>
+  </item>
 </rdf:RDF>
