diff -Nur -x '*.orig' -x '*~' system-config-printer-1.5.3+20141029/config.py.in system-config-printer-1.5.3+20141029.new/config.py.in
--- system-config-printer-1.5.3+20141029/config.py.in	2014-10-29 19:44:59.000000000 +0100
+++ system-config-printer-1.5.3+20141029.new/config.py.in	2014-12-04 23:28:49.093199899 +0100
@@ -24,7 +24,7 @@
 pkgdatadir="@datadir@/@PACKAGE@"
 VERSION="@VERSION@"
 PACKAGE="@PACKAGE@"
-DOWNLOADABLE_ONLYPPD=True
-DOWNLOADABLE_ONLYFREE=True
+DOWNLOADABLE_ONLYPPD=False
+DOWNLOADABLE_ONLYFREE=False
 DOWNLOADABLE_PKG_ONLYSIGNED=True
 packagesystem = None # discovered at runtime
