diff -Nur -x '*.orig' -x '*~' system-config-printer-1.3.8+20120201/print-applet.desktop.in system-config-printer-1.3.8+20120201.new/print-applet.desktop.in
--- system-config-printer-1.3.8+20120201/print-applet.desktop.in	2012-02-16 17:05:25.850598002 +0100
+++ system-config-printer-1.3.8+20120201.new/print-applet.desktop.in	2012-02-16 17:08:52.250598002 +0100
@@ -6,7 +6,7 @@
 Terminal=false
 Type=Application
 Icon=printer
-NotShowIn=KDE;LXDE;
+NotShowIn=KDE;LXDE;Unity;GNOME;
 StartupNotify=false
 X-GNOME-Autostart-Delay=30
 X-Ubuntu-Gettext-Domain=system-config-printer
