diff -Nur -x '*.orig' -x '*~' system-config-printer-1.5.7+20150819/printerproperties.py system-config-printer-1.5.7+20150819.new/printerproperties.py
--- system-config-printer-1.5.7+20150819/printerproperties.py	2015-08-19 00:11:18.000000000 -0300
+++ system-config-printer-1.5.7+20150819.new/printerproperties.py	2015-10-03 09:14:32.741612266 -0300
@@ -1229,7 +1229,7 @@
             # Printer has been deleted meanwhile
             return
 
-        with tempfile.NamedTemporaryFile(mode='wt') as tmpfile:
+        with tempfile.NamedTemporaryFile(mode='wt', buffering=1) as tmpfile:
             tmpfile.write ("#CUPS-COMMAND\n%s\n" % command)
             self.cups._begin_operation (_("sending maintenance command"))
             try:
