--- a/data/epiphany.desktop.in
+++ b/data/epiphany.desktop.in
@@ -11,7 +11,11 @@
 Icon=web-browser
 Categories=Network;GNOME;GTK;WebBrowser;
 MimeType=text/html;text/xml;application/xhtml+xml;x-scheme-handler/http;x-scheme-handler/https;multipart/related;application/x-mimearchive;message/rfc822;
-Actions=Incognito;
+Actions=New;Incognito;
+
+[Desktop Action New]
+_Name=New Window
+Exec=epiphany --new-window
 
 [Desktop Action Incognito]
 _Name=New Incognito Window
