--- a/src/resources/prefs-dialog.ui
+++ b/src/resources/prefs-dialog.ui
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!-- Generated with glade 3.16.0 -->
+<!-- Generated with glade 3.18.3 -->
 <interface>
   <requires lib="gtk+" version="3.0"/>
   <object class="GtkAdjustment" id="adjustment1">
@@ -18,7 +18,7 @@
     <columns>
       <!-- column-name name -->
       <column type="gchararray"/>
-      <!-- column-name stock url -->
+      <!-- column-name stock -->
       <column type="gchararray"/>
       <!-- column-name url -->
       <column type="gchararray"/>
@@ -33,24 +33,14 @@
     <property name="destroy_with_parent">True</property>
     <property name="icon_name">gtk-preferences</property>
     <property name="type_hint">dialog</property>
-    <child internal-child="headerbar">
-      <object class="GtkHeaderBar" id="headerbar1">
-        <property name="title" translatable="yes">Preferences</property>
-        <property name="visible">True</property>
-        <property name="can_focus">False</property>
-        <property name="show-close-button">True</property>
-      </object>
-    </child>
     <child internal-child="vbox">
       <object class="GtkBox" id="dialog-vbox2">
         <property name="visible">True</property>
         <property name="can_focus">False</property>
-        <property name="border_width">0</property>
         <property name="orientation">vertical</property>
         <property name="spacing">2</property>
         <child internal-child="action_area">
           <object class="GtkButtonBox" id="dialog-action_area2">
-            <property name="visible">False</property>
             <property name="can_focus">False</property>
             <property name="layout_style">end</property>
           </object>
@@ -65,8 +55,10 @@
           <object class="GtkNotebook" id="prefs_notebook">
             <property name="visible">True</property>
             <property name="can_focus">True</property>
+            <property name="margin_left">12</property>
+            <property name="margin_right">12</property>
+            <property name="margin_bottom">12</property>
             <property name="border_width">0</property>
-            <property name="show_border">False</property>
             <child>
               <object class="GtkBox" id="general_page_box">
                 <property name="visible">True</property>
@@ -106,14 +98,13 @@
                           <object class="GtkBox" id="vbox197">
                             <property name="visible">True</property>
                             <property name="can_focus">False</property>
+                            <property name="margin_start">12</property>
                             <property name="orientation">vertical</property>
                             <property name="spacing">6</property>
-                            <property name="margin_start">12</property>
                             <child>
                               <object class="GtkBox" id="download_button_hbox">
                                 <property name="visible">True</property>
                                 <property name="can_focus">False</property>
-                                <property name="orientation">horizontal</property>
                                 <property name="spacing">12</property>
                                 <child>
                                   <object class="GtkLabel" id="download_button_label">
@@ -145,6 +136,7 @@
                                 <property name="can_focus">True</property>
                                 <property name="receives_default">False</property>
                                 <property name="use_underline">True</property>
+                                <property name="xalign">0.5</property>
                                 <property name="draw_indicator">True</property>
                               </object>
                               <packing>
@@ -200,9 +192,9 @@
                       <object class="GtkBox" id="vbox7">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
+                        <property name="margin_start">12</property>
                         <property name="orientation">vertical</property>
                         <property name="spacing">6</property>
-                        <property name="margin_start">12</property>
                         <child>
                           <object class="GtkBox" id="search_engine_hbox1">
                             <property name="visible">True</property>
@@ -281,17 +273,18 @@
                       <object class="GtkBox" id="vbox186">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
+                        <property name="margin_start">12</property>
                         <property name="orientation">vertical</property>
                         <property name="spacing">6</property>
-                        <property name="margin_start">12</property>
                         <child>
                           <object class="GtkCheckButton" id="popups_allow_checkbutton">
                             <property name="label" translatable="yes">Allow popup _windows</property>
                             <property name="visible">True</property>
                             <property name="can_focus">True</property>
                             <property name="receives_default">False</property>
-                            <property name="use_underline">True</property>
                             <property name="halign">start</property>
+                            <property name="use_underline">True</property>
+                            <property name="xalign">0.5</property>
                             <property name="draw_indicator">True</property>
                           </object>
                           <packing>
@@ -306,8 +299,9 @@
                             <property name="visible">True</property>
                             <property name="can_focus">True</property>
                             <property name="receives_default">False</property>
-                            <property name="use_underline">True</property>
                             <property name="halign">start</property>
+                            <property name="use_underline">True</property>
+                            <property name="xalign">0.5</property>
                             <property name="draw_indicator">True</property>
                           </object>
                           <packing>
@@ -322,8 +316,9 @@
                             <property name="visible">True</property>
                             <property name="can_focus">True</property>
                             <property name="receives_default">False</property>
-                            <property name="use_underline">True</property>
                             <property name="halign">start</property>
+                            <property name="use_underline">True</property>
+                            <property name="xalign">0.5</property>
                             <property name="draw_indicator">True</property>
                           </object>
                           <packing>
@@ -368,8 +363,8 @@
                 <child>
                   <object class="GtkBox" id="vbox1">
                     <property name="visible">True</property>
-                    <property name="orientation">vertical</property>
                     <property name="can_focus">False</property>
+                    <property name="orientation">vertical</property>
                     <child>
                       <object class="GtkLabel" id="label1">
                         <property name="visible">True</property>
@@ -390,9 +385,9 @@
                       <object class="GtkBox" id="vbox2">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
+                        <property name="margin_start">12</property>
                         <property name="orientation">vertical</property>
                         <property name="spacing">6</property>
-                        <property name="margin_start">12</property>
                         <child>
                           <object class="GtkCheckButton" id="use_gnome_fonts_checkbutton">
                             <property name="label" translatable="yes">_Use system fonts</property>
@@ -400,6 +395,7 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">False</property>
                             <property name="use_underline">True</property>
+                            <property name="xalign">0.5</property>
                             <property name="draw_indicator">True</property>
                           </object>
                           <packing>
@@ -412,11 +408,11 @@
                           <object class="GtkTable" id="custom_fonts_table">
                             <property name="visible">True</property>
                             <property name="can_focus">False</property>
+                            <property name="margin_start">12</property>
                             <property name="n_rows">3</property>
                             <property name="n_columns">2</property>
                             <property name="column_spacing">12</property>
                             <property name="row_spacing">6</property>
-                            <property name="margin_start">12</property>
                             <child>
                               <object class="GtkFontButton" id="sans_fontbutton">
                                 <property name="visible">True</property>
@@ -549,9 +545,8 @@
                       <object class="GtkBox" id="css_container">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
-                        <property name="orientation">horizontal</property>
-                        <property name="spacing">24</property>
                         <property name="margin_start">12</property>
+                        <property name="spacing">24</property>
                         <child>
                           <object class="GtkCheckButton" id="css_checkbox">
                             <property name="label" translatable="yes">Use custom _stylesheet</property>
@@ -559,6 +554,7 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">False</property>
                             <property name="use_underline">True</property>
+                            <property name="xalign">0.5</property>
                             <property name="draw_indicator">True</property>
                           </object>
                           <packing>
@@ -684,9 +680,9 @@
                       <object class="GtkBox" id="vbox188">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
+                        <property name="margin_start">12</property>
                         <property name="orientation">vertical</property>
                         <property name="spacing">6</property>
-                        <property name="margin_start">12</property>
                         <child>
                           <object class="GtkRadioButton" id="always">
                             <property name="label" translatable="yes">_Always accept</property>
@@ -694,6 +690,7 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">False</property>
                             <property name="use_underline">True</property>
+                            <property name="xalign">0.5</property>
                             <property name="draw_indicator">True</property>
                           </object>
                           <packing>
@@ -709,6 +706,7 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">False</property>
                             <property name="use_underline">True</property>
+                            <property name="xalign">0.5</property>
                             <property name="draw_indicator">True</property>
                             <property name="group">always</property>
                           </object>
@@ -723,9 +721,9 @@
                             <property name="visible">True</property>
                             <property name="can_focus">False</property>
                             <property name="halign">start</property>
+                            <property name="margin_start">22</property>
                             <property name="label" translatable="yes" comments="Refers to &quot;Only from sites you visit&quot; option under Cookies.">&lt;small&gt;For example, not from advertisers on these sites&lt;/small&gt;</property>
                             <property name="use_markup">True</property>
-                            <property name="margin_start">22</property>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -740,6 +738,7 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">False</property>
                             <property name="use_underline">True</property>
+                            <property name="xalign">0.5</property>
                             <property name="draw_indicator">True</property>
                             <property name="group">always</property>
                           </object>
@@ -791,10 +790,11 @@
                         <property name="visible">True</property>
                         <property name="can_focus">True</property>
                         <property name="receives_default">False</property>
-                        <property name="use_underline">True</property>
                         <property name="halign">start</property>
-                        <property name="draw_indicator">True</property>
                         <property name="margin_start">12</property>
+                        <property name="use_underline">True</property>
+                        <property name="xalign">0.5</property>
+                        <property name="draw_indicator">True</property>
                       </object>
                       <packing>
                         <property name="expand">False</property>
@@ -864,9 +864,10 @@
                         <property name="visible">True</property>
                         <property name="can_focus">True</property>
                         <property name="receives_default">False</property>
+                        <property name="margin_start">12</property>
                         <property name="use_underline">True</property>
+                        <property name="xalign">0.5</property>
                         <property name="draw_indicator">True</property>
-                        <property name="margin_start">12</property>
                       </object>
                       <packing>
                         <property name="expand">False</property>
@@ -907,27 +908,8 @@
                       <object class="GtkBox" id="hbox174">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
-                        <property name="orientation">horizontal</property>
-                        <property name="spacing">12</property>
                         <property name="margin_start">12</property>
-                        <child>
-                          <object class="GtkLabel">
-                            <property name="visible">True</property>
-                            <property name="can_focus">False</property>
-                            <!-- We must use xalign instead of halign because halign
-                                 does not work for labels with line wrap. And we must
-                                 allow line wrap to accomodate translations. -->
-                            <property name="xalign">0</property>
-                            <property name="wrap">True</property>
-                            <property name="max-width-chars">30</property>
-                            <property name="label" translatable="yes">You can clear stored personal data.</property>
-                          </object>
-                          <packing>
-                            <property name="expand">True</property>
-                            <property name="fill">True</property>
-                            <property name="position">0</property>
-                          </packing>
-                        </child>
+                        <property name="spacing">12</property>
                         <child>
                           <object class="GtkButton" id="clear_personal_data_button">
                             <property name="label" translatable="yes">Cl_ear Personal Data…</property>
@@ -939,8 +921,8 @@
                           <packing>
                             <property name="expand">False</property>
                             <property name="fill">False</property>
-                            <property name="position">1</property>
                             <property name="pack_type">end</property>
+                            <property name="position">1</property>
                           </packing>
                         </child>
                       </object>
@@ -1006,10 +988,10 @@
                       <object class="GtkTable" id="table72">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
+                        <property name="margin_start">12</property>
                         <property name="n_columns">2</property>
                         <property name="column_spacing">12</property>
                         <property name="row_spacing">6</property>
-                        <property name="margin_start">12</property>
                         <child>
                           <object class="GtkLabel" id="default_encoding_label">
                             <property name="visible">True</property>
@@ -1076,10 +1058,10 @@
                       <object class="GtkGrid" id="grid1">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
+                        <property name="margin_start">12</property>
                         <property name="vexpand">True</property>
                         <property name="row_spacing">6</property>
                         <property name="column_spacing">6</property>
-                        <property name="margin_start">12</property>
                         <child>
                           <object class="GtkScrolledWindow" id="scrolledwindow2">
                             <property name="visible">True</property>
@@ -1102,8 +1084,6 @@
                           <packing>
                             <property name="left_attach">0</property>
                             <property name="top_attach">0</property>
-                            <property name="width">1</property>
-                            <property name="height">1</property>
                           </packing>
                         </child>
                         <child>
@@ -1176,8 +1156,6 @@
                           <packing>
                             <property name="left_attach">1</property>
                             <property name="top_attach">0</property>
-                            <property name="width">1</property>
-                            <property name="height">1</property>
                           </packing>
                         </child>
                       </object>
@@ -1217,17 +1195,18 @@
                         <property name="position">0</property>
                       </packing>
                     </child>
-                      <child>
-                        <object class="GtkCheckButton" id="enable_spell_checking_checkbutton">
-                          <property name="label" translatable="yes">_Enable spell checking</property>
-                          <property name="visible">True</property>
-                          <property name="can_focus">True</property>
-                          <property name="receives_default">False</property>
-                          <property name="use_underline">True</property>
-                          <property name="halign">start</property>
-                          <property name="draw_indicator">True</property>
-                          <property name="margin_start">12</property>
-                        </object>
+                    <child>
+                      <object class="GtkCheckButton" id="enable_spell_checking_checkbutton">
+                        <property name="label" translatable="yes">_Enable spell checking</property>
+                        <property name="visible">True</property>
+                        <property name="can_focus">True</property>
+                        <property name="receives_default">False</property>
+                        <property name="halign">start</property>
+                        <property name="margin_start">12</property>
+                        <property name="use_underline">True</property>
+                        <property name="xalign">0.5</property>
+                        <property name="draw_indicator">True</property>
+                      </object>
                       <packing>
                         <property name="expand">False</property>
                         <property name="fill">True</property>
@@ -1267,12 +1246,4 @@
       </object>
     </child>
   </template>
-  <object class="GtkSizeGroup">
-    <property name="mode">horizontal</property>
-    <widgets>
-      <widget name="manage_cookies_button"/>
-      <widget name="manage_passwords_button"/>
-      <widget name="clear_personal_data_button"/>
-    </widgets>
-  </object>
 </interface>
