--- a/src/resources/epiphany-application-menu.ui
+++ b/src/resources/epiphany-application-menu.ui
@@ -8,11 +8,6 @@
         <attribute name="accel">&lt;Primary&gt;n</attribute>
       </item>
       <item>
-        <attribute name="label" translatable="yes">New _Incognito Window</attribute>
-        <attribute name="action">app.new-incognito</attribute>
-        <attribute name="accel">&lt;Primary&gt;i</attribute>
-      </item>
-      <item>
         <attribute name="label" translatable="yes">Reopen Closed _Tab</attribute>
         <attribute name="action">app.reopen-closed-tab</attribute>
         <attribute name="accel">&lt;Primary&gt;&lt;Shift&gt;t</attribute>
