--- a/src/ephy-toolbar.c
+++ b/src/ephy-toolbar.c
@@ -166,6 +166,22 @@
   gtk_image_set_pixel_size (gtk_button_get_image (GTK_BUTTON (button)), 24);
   gtk_header_bar_pack_start (GTK_HEADER_BAR (toolbar), button);
 
+  /* Home */
+  button = ephy_middle_clickable_button_new ();
+  /* FIXME: apparently we need an image inside the button for the action
+   * icon to appear. */
+  gtk_button_set_image (GTK_BUTTON (button), gtk_image_new ());
+  gtk_widget_set_valign (button, GTK_ALIGN_CENTER);
+  action_group = ephy_window_get_toolbar_action_group (priv->window);
+  action = gtk_action_group_get_action (action_group, "FileHome");
+  gtk_activatable_set_related_action (GTK_ACTIVATABLE (button),
+                                      action);
+  gtk_button_set_label (GTK_BUTTON (button), NULL);
+  gtk_activatable_set_use_action_appearance (GTK_ACTIVATABLE (button), FALSE);
+  gtk_button_set_image (GTK_BUTTON (button), gtk_image_new_from_icon_name ("go-home-symbolic", GTK_ICON_SIZE_LARGE_TOOLBAR));
+  gtk_style_context_add_class (gtk_widget_get_style_context (button), "image-button");
+  gtk_header_bar_pack_start (GTK_HEADER_BAR (toolbar), button);
+
   /* New Tab */
   button = gtk_button_new ();
   priv->new_tab_button = button;
