--- a/src/ephy-toolbar.c
+++ b/src/ephy-toolbar.c
@@ -163,6 +163,7 @@
   action = gtk_action_group_get_action (action_group, "ViewCombinedStopReload");
   gtk_activatable_set_related_action (GTK_ACTIVATABLE (button),
                                       action);
+  gtk_widget_set_tooltip_text (button, gtk_action_get_tooltip (action));
   gtk_image_set_pixel_size (gtk_button_get_image (GTK_BUTTON (button)), 24);
   gtk_header_bar_pack_start (GTK_HEADER_BAR (toolbar), button);
 
@@ -177,6 +178,7 @@
   gtk_activatable_set_related_action (GTK_ACTIVATABLE (button),
                                       action);
   gtk_button_set_label (GTK_BUTTON (button), NULL);
+  gtk_widget_set_tooltip_text (button, gtk_action_get_label (action));
   gtk_activatable_set_use_action_appearance (GTK_ACTIVATABLE (button), FALSE);
   gtk_button_set_image (GTK_BUTTON (button), gtk_image_new_from_icon_name ("go-home-symbolic", GTK_ICON_SIZE_LARGE_TOOLBAR));
   gtk_style_context_add_class (gtk_widget_get_style_context (button), "image-button");
