# DP: document readline header location on Debian systems

Index: b/examples/loadables/README
===================================================================
--- a/examples/loadables/README
+++ b/examples/loadables/README
@@ -38,6 +38,9 @@ rest of the example builtins. It's inten
 that can be modified or included to help you build your own loadables
 without having to search for the right CFLAGS and LDFLAGS.
 
+On Debian GNU/Linux systems, the bash headers are in /usr/include/bash.
+The appropriate options are already set in the example Makefile.
+
 basename.c	Return non-directory portion of pathname.
 cat.c		cat(1) replacement with no options - the way cat was intended.
 dirname.c	Return directory portion of pathname.
