From: Dmitrijs Ledkovs <xnox@ubuntu.com>
Date: Tue, 2 Jan 2024 10:23:29 +0100
Subject: Implement update status (U) splash plugin function for details
 plugins

Bug-Ubuntu: https://bugs.launchpad.net/bugs/540645
---
 src/plugins/splash/details/plugin.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/plugins/splash/details/plugin.c b/src/plugins/splash/details/plugin.c
index d1543ed..3875f07 100644
--- a/src/plugins/splash/details/plugin.c
+++ b/src/plugins/splash/details/plugin.c
@@ -80,6 +80,9 @@ struct _ply_boot_splash_plugin
         ply_buffer_t                  *boot_buffer;
 };
 
+static void display_message (ply_boot_splash_plugin_t *plugin,
+                             const char               *message);
+
 static view_t *
 view_new (ply_boot_splash_plugin_t *plugin,
           ply_text_display_t       *display)
@@ -325,6 +328,19 @@ update_status (ply_boot_splash_plugin_t *plugin,
         assert (plugin != NULL);
 
         ply_trace ("status update");
+
+        int progress = 0;
+
+        if (! strncmp("fsck:", status, 5)) {
+                /* Chop localised formatted string */
+                /* fsck:sda1:50:Checking disk %1$d of %2$d (%3$d%% complete) */
+                sscanf(status, "fsck:.*:%d:.*", &progress);
+                char *end = strrchr(status, ':');
+                strncpy (end, "%\0", 2);
+        }
+
+        if (progress < 100)
+               display_message (plugin, status);
 }
 
 static void
