From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Sun, 18 Feb 2018 15:22:45 -0500
Subject: Disable tests that require Internet access

---
 tests/meson.build | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/meson.build b/tests/meson.build
index 89f903a..b9c3f7d 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -78,10 +78,10 @@ if get_option('unit_tests')
     'ephy-gsb-service-test.c',
     dependencies: ephymain_dep
   )
-  test('GSB service test',
-       gsb_service_test,
-       env: envs
-  )
+  # test('GSB service test',
+  #      gsb_service_test,
+  #      env: envs
+  # )
 
   history_test = executable('test-ephy-history',
     'ephy-history-test.c',
