diff --git a/debian/grub.d/05_debian_theme b/debian/grub.d/05_debian_theme
index 717258444..3d63630f9 100755
--- a/debian/grub.d/05_debian_theme
+++ b/debian/grub.d/05_debian_theme
@@ -44,9 +44,9 @@ set_default_theme(){
                        fi
                        ;;
                *)
-                       # Set the traditional Debian blue theme.
-                       echo "${1}set menu_color_normal=cyan/blue"
-                       echo "${1}set menu_color_highlight=white/blue"
+                       # Set the traditional elementary black and white theme.
+                       echo "${1}set menu_color_normal=white/black"
+                       echo "${1}set menu_color_highlight=black/white"
                        ;;
        esac
 }

