Description: expect zst in tests
Author: Dimitri John Ledkov <dimitri.ledkov@canonical.com>

Index: dkms-3.0.11/run_test.sh
===================================================================
--- dkms-3.0.11.orig/run_test.sh
+++ dkms-3.0.11/run_test.sh
@@ -244,9 +244,13 @@ case "${os_id}" in
     sles | suse | opensuse)
         expected_dest_loc=updates
         ;;
-    arch | debian | ubuntu)
+    arch | debian)
         expected_dest_loc=updates/dkms
         ;;
+    ubuntu)
+        expected_dest_loc=updates/dkms
+        mod_compression_ext=.zst
+        ;;
     alpine)
         expected_dest_loc=kernel/extra
         mod_compression_ext=.gz
