=== modified file 'src/applet.c'
--- old/src/applet.c	2017-05-09 13:47:30 +0000
+++ new/src/applet.c	2017-05-09 13:49:00 +0000
@@ -2465,6 +2465,13 @@
 		applet->permissions[permission] = result;
 }
 
+static void
+foo_wireless_enabled_changed_cb (NMClient *client, GParamSpec *pspec, NMApplet *applet)
+{
+	applet_schedule_update_icon (applet);
+	applet_schedule_update_menu (applet);
+}
+
 static gboolean
 foo_set_initial_state (gpointer data)
 {
@@ -2514,6 +2521,14 @@
 	                  G_CALLBACK (foo_manager_permission_changed),
 	                  applet);
 
+	g_signal_connect (applet->nm_client, "notify::wireless-enabled",
+	                  G_CALLBACK (foo_wireless_enabled_changed_cb),
+	                  applet);
+
+	g_signal_connect (applet->nm_client, "notify::wwan-enabled",
+	                  G_CALLBACK (foo_wireless_enabled_changed_cb),
+	                  applet);
+
 	/* Initialize permissions - the initial 'permission-changed' signal is emitted from NMClient constructor, and thus not caught */
 	for (perm = NM_CLIENT_PERMISSION_NONE + 1; perm <= NM_CLIENT_PERMISSION_LAST; perm++) {
 		applet->permissions[perm] = nm_client_get_permission_result (applet->nm_client, perm);

