Description: fix compatibility with newer WebKitGTK+.
Origin: upstream, https://git.gnome.org/browse/epiphany/commit/?h=gnome-3-18&id=8ea593d0a26679fcb1d94d25be9d0235a0a67668
Origin: upstream, https://git.gnome.org/browse/epiphany/commit/?h=gnome-3-18&id=611ece6ac03caeaad7be1f6c5c1129f80aa2d7f4

Index: epiphany-browser-3.18.5/configure.ac
===================================================================
--- epiphany-browser-3.18.5.orig/configure.ac	2017-01-09 08:14:11.000000000 -0500
+++ epiphany-browser-3.18.5/configure.ac	2017-01-10 09:33:37.790858462 -0500
@@ -77,7 +77,7 @@
 GTK_REQUIRED=3.13.0
 LIBXML_REQUIRED=2.6.12
 LIBXSLT_REQUIRED=1.1.7
-WEBKIT_GTK_REQUIRED=2.9.5
+WEBKIT_GTK_REQUIRED=2.13.2
 LIBSOUP_REQUIRED=2.48.0
 GNOME_DESKTOP_REQUIRED=2.91.2
 LIBSECRET_REQUIRED=0.14
Index: epiphany-browser-3.18.5/embed/web-extension/ephy-web-overview.c
===================================================================
--- epiphany-browser-3.18.5.orig/embed/web-extension/ephy-web-overview.c	2016-03-14 19:15:49.000000000 -0400
+++ epiphany-browser-3.18.5/embed/web-extension/ephy-web-overview.c	2017-01-10 09:33:37.790858462 -0500
@@ -144,8 +144,8 @@
       item->url = g_strdup (url->url);
 
       class_list = webkit_dom_element_get_class_list (webkit_dom_node_get_parent_element (WEBKIT_DOM_NODE (item->anchor)));
-      if (class_list && webkit_dom_dom_token_list_contains (class_list, "overview-removed", NULL))
-        webkit_dom_dom_token_list_remove (class_list, "overview-removed", NULL);
+      if (class_list && webkit_dom_dom_token_list_contains (class_list, "overview-removed"))
+        webkit_dom_dom_token_list_remove (class_list, NULL, "overview-removed", NULL);
 
       webkit_dom_element_set_attribute (item->anchor, "href", url->url, NULL);
       webkit_dom_element_set_attribute (item->anchor, "title", url->title, NULL);
