From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Force a menu update 30 seconds after the initialization is done.

---
 src/applet.c |    2 ++
 1 file changed, 2 insertions(+)

Index: network-manager-applet-1.0.4/src/applet.c
===================================================================
--- network-manager-applet-1.0.4.orig/src/applet.c
+++ network-manager-applet-1.0.4/src/applet.c
@@ -3914,6 +3914,8 @@ setup_indicator_menu (NMApplet *applet)
 
 	app_indicator_set_menu(applet->app_indicator, GTK_MENU(applet->menu));
 
+	applet_schedule_update_menu (applet);
+
 	return TRUE;
 }
 #endif /* ENABLE_INDICATOR */
