diff --git a/tests/testlibrary.c b/tests/testlibrary.c
index 4b3e11fb..195bcab8 100644
--- a/tests/testlibrary.c
+++ b/tests/testlibrary.c
@@ -2611,6 +2611,8 @@ global_setup (void)
   g_setenv ("FLATPAK_SYSTEM_HELPER_ON_SESSION", "1", TRUE);
   g_setenv ("LANGUAGE", "en", TRUE);
 
+  g_setenv ("FLATPAK_DISABLE_REVOKEFS", "1", TRUE);
+
   test_bus = g_test_dbus_new (G_TEST_DBUS_NONE);
 
   services_dir = g_test_build_filename (G_TEST_BUILT, "services", NULL);

