Description: add raring version number as a known release
Origin: backport, http://live.debian.net/gitweb/?p=live-build.git;a=commit;h=d766063e128b6f51df58e90956f462343ca3808d
Bug-Ubuntu: https://launchpad.net/bugs/1174791
Forwarded: not-needed
---
Index: live-build-3.0~a57/functions/releases.sh
===================================================================
--- live-build-3.0~a57.orig/functions/releases.sh	2012-08-09 20:32:51.000000000 +0900
+++ live-build-3.0~a57/functions/releases.sh	2013-05-01 00:42:31.044073697 +0900
@@ -26,3 +26,4 @@
 RELEASE_oneiric="11.10"		# 2013-04-01
 RELEASE_precise="12.04"		# LTS: 2017-04-01
 RELEASE_quantal="12.10"		# 2014-04-01
+RELEASE_raring="13.04"		# 2014-01-01
