Description: Use i386 -generic-pae kernel flavour on Ubuntu precise
Author: Colin Watson <cjwatson@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/897786
Forwarded: no
Last-Update: 2012-08-20

Index: b/functions/defaults.sh
===================================================================
--- a/functions/defaults.sh
+++ b/functions/defaults.sh
@@ -748,7 +748,14 @@
 					;;
 
 				ubuntu|kubuntu)
-					LB_LINUX_FLAVOURS="${LB_LINUX_FLAVOURS:-generic}"
+					case "${LB_DISTRIBUTION}" in
+						precise)
+							LB_LINUX_FLAVOURS="${LB_LINUX_FLAVOURS:-generic-pae}"
+							;;
+						*)
+							LB_LINUX_FLAVOURS="${LB_LINUX_FLAVOURS:-generic}"
+							;;
+					esac
 					;;
 
 				*)
Index: b/scripts/build/lb_binary_debian-installer
===================================================================
--- a/scripts/build/lb_binary_debian-installer
+++ b/scripts/build/lb_binary_debian-installer
@@ -362,7 +362,14 @@
 
 			case "${LB_MODE}" in
 				ubuntu|kubuntu)
-					DI_PACKAGES="${DI_REQ_PACKAGES} linux-generic"
+					case "${LB_DISTRIBUTION}" in
+						precise)
+							DI_PACKAGES="${DI_REQ_PACKAGES} linux-generic-pae"
+							;;
+						*)
+							DI_PACKAGES="${DI_REQ_PACKAGES} linux-generic"
+							;;
+					esac
 					;;
 
 				*)
