Index: b/doc/Makefile.in
===================================================================
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -70,8 +70,8 @@
 TEX	    = tex
 
 MAKEINFO    = makeinfo
-TEXI2DVI    = ${SUPPORT_SRCDIR}/texi2dvi
-TEXI2HTML   = ${SUPPORT_SRCDIR}/texi2html
+TEXI2DVI    = texi2dvi
+TEXI2HTML   = texi2html
 MAN2HTML    = ${BUILD_DIR}/support/man2html
 HTMLPOST    = ${srcdir}/htmlpost.sh
 INFOPOST    = ${srcdir}/infopost.sh
