--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -66,8 +66,8 @@ SUPPORT_SRCDIR = $(topdir)/support
 RL_LIBDIR = $(topdir)/lib/readline
 
 MAKEINFO    = makeinfo
-TEXI2DVI    = ${SUPPORT_SRCDIR}/texi2dvi
-TEXI2HTML   = ${SUPPORT_SRCDIR}/texi2html
+TEXI2DVI    = texi2dvi
+TEXI2HTML   = texi2html
 MAN2HTML    = ${BUILD_DIR}/support/man2html
 HTMLPOST    = ${srcdir}/htmlpost.sh
 QUIETPS	    = #set this to -q to shut up dvips
